### PR TITLE
Add Backup test and enable semi-sync in the tests

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This Pull request does the following 3 things - 
  - Refactors the current upgrade end to end test to extract out the helper functions into a `utils.sh` file.
  - Alter the test to setup `semi-sync` on the vttablets. For this purpose the number of replicas were increased from 1 to 2.
  - Adds a new backup-restore test which first takes a backup of the shard, followed by a restoration of the entire shard using that backup via the operator.

### CODE REVIEW

File: test/endtoend/backup_restore_test.sh [hunk 1]
```diff
@@ -0,0 +1,196 @@
+#!/bin/bash
+
+source ./tools/test.env
+source ./test/endtoend/utils.sh
+
+function get_started() {
+    echo "Apply latest operator.yaml"
```
Comment (frouioui) [inline test/endtoend/backup_restore_test.sh:7]: nit: I think `get_started()` can be shared between `backup_restore_test` and `upgrade_test`. The initial cluster manifest we want to apply could be passed down as an argument to `get_started()`

File: test/endtoend/operator/102_keyspace_teardown.yaml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+type: Opaque
+stringData:
+  users.json: |
+    {
+      "user": [{
+        "UserData": "user",
+        "Password": ""
+      }]
+    }
+  init_db.sql: |
```
Comment (frouioui) [inline test/endtoend/operator/102_keyspace_teardown.yaml:70]: nit: this whole block can be removed

**Review by frouioui — APPROVED** [review]
Looks good to me! I left a few nits. Great work @GuptaManan100

File: test/endtoend/operator/102_keyspace_teardown.yaml (hunk 2 above)
Comment (GuptaManan100) [reply test/endtoend/operator/102_keyspace_teardown.yaml]: Fixed

File: test/endtoend/backup_restore_test.sh (hunk 1 above)
Comment (GuptaManan100) [reply test/endtoend/backup_restore_test.sh]: Fixed!

**Pull request merged**